### DISCUSSION TITLE
Filter by relation length

### DISCUSSION THREAD

**Discussion by elliotsoomro** [pocketbase/pocketbase#1009] [Q&A] [answered]
Is it possible to filter records by the number of selected records in a relation field? E.g. filter "questions" by how many "alternatives" they have, such as only return questions with >0 alternatives.

This does not seem to work: filter="alternatives>0"

**Comment by ganigeorgiev** [comment, marked as answer]
No, checking the array value length at the moment is not supported by the filter expressions.

As a workaround for your use case, you could try comparing the `alternatives` relation field to a non empty string or array, or even checking if there is an id of at least 1 alternative item, eg.:

```
// option 1 (depends on what is the maxSelect for your relation)
alternatives != ""

// option 2 (depends on what is the maxSelect for your relation)
altenatives != "[]"

// option 3 (should work for both single and multiple relations)
alternatives.id != ""
```
